File: packages/flarelette-jwt-py/prepare.py
#!/usr/bin/env python3
import shutil
import sys
from pathlib import Path

PKG_DIR = Path(__file__).resolve().parent
MD_ROOT = (PKG_DIR / "../..").resolve()

README_SRC = MD_ROOT / "README.md"
CONTRIB_SRC = MD_ROOT / "CONTRIBUTING.md"
THIRD_PARTY_SRC = MD_ROOT / "THIRD_PARTY_LICENSES.md"
LICENSE_SRC = MD_ROOT / "LICENSE"
DOCS_DIR = PKG_DIR
DOCS_DIR.mkdir(parents=True, exist_ok=True)

print(f"Copying documentation files from {MD_ROOT} to {DOCS_DIR} directory...")
for src, dst_name in [
    (README_SRC, "README.md"),
    (CONTRIB_SRC, "CONTRIBUTING.md"),
    (THIRD_PARTY_SRC, "THIRD_PARTY_LICENSES.md"),
    (LICENSE_SRC, "LICENSE"),
]:
    if not src.is_file():
        print(f"ERROR: Missing {src}", file=sys.stderr)
        sys.exit(1)
    print(f"Copying {dst_name}...")
    shutil.copy2(src, DOCS_DIR / dst_name)
print("Copy complete")

File: packages/flarelette-jwt-ts/prepare.js
#!/usr/bin/env node
import { copyFileSync, existsSync } from 'fs'
import { dirname, join, resolve } from 'path'
import { fileURLToPath } from 'url'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)

const PKG_DIR = resolve(__dirname)
const MD_ROOT = resolve(PKG_DIR, '../..')

const files = [
  { src: join(MD_ROOT, 'README.md'), dst: 'README.md' },
  { src: join(MD_ROOT, 'CONTRIBUTING.md'), dst: 'CONTRIBUTING.md' },
  { src: join(MD_ROOT, 'THIRD_PARTY_LICENSES.md'), dst: 'THIRD_PARTY_LICENSES.md' },
  { src: join(MD_ROOT, 'LICENSE'), dst: 'LICENSE' },
]

console.log(`Copying documentation files from ${MD_ROOT} to ${PKG_DIR} directory...`)

for (const { src, dst } of files) {
  if (!existsSync(src)) {
    console.error(`ERROR: Missing ${src}`)
    process.exit(1)
  }
  console.log(`Copying ${dst}...`)
  copyFileSync(src, join(PKG_DIR, dst))
}

console.log('Copy complete')
